ElectricCar.cpp

```diff
@@ -4,13 +4,48 @@
 ElectricCar::ElectricCar(ElectricEngine* engine)
     : engine_(engine)
 {
-    std::cout << __FUNCTION__ << std::endl;
+    std::cout << "* " << __FUNCTION__ << std::endl;
 }
 
-ElectricCar::~ElectricCar()       { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::charge()        { std::cout << __FUNCTION__ << std::endl; }
+ElectricCar::~ElectricCar() {
+    std::cout << "* " << __FUNCTION__ << std::endl;
+    delete engine_;
+}
+
+void ElectricCar::charge() { 
+    std::cout << __FUNCTION__;
+    if (chargeLevel_ < MAX_CAPACITY) {
+        chargeLevel_ = MAX_CAPACITY;
+        std::cout << " : recharged" << std::endl;
+    } else {
+        std::cerr << " : battery already full" << std::endl;
+    }
+}
+
+size_t ElectricCar::getChargeLevel() {
+    return chargeLevel_;
+}
 
+void ElectricCar::setEngine(ElectricEngine* engine) {
+    std::cout << __FUNCTION__;
+    if (!getSpeed()) {
```

Comment:
it's better to check int values explicitly with ==, not with negation.

Reply:
Ok. Done.